src/codegen/sys/lib_.rs

```diff
@@ -23,6 +23,26 @@ pub fn generate(env: &Env) {
     save_to_file(&path, env.config.make_backup, |w| generate_lib(w, env));
 }
 
+fn get_library_name(lib: &String) -> String {
+    let mut s = String::from(lib);
+
+    if s.starts_with("lib") {
+        s = s.split_off("lib".len());
+    }
+
+    let offset = s.find(".so").unwrap_or(s.len());
+    s.truncate(offset);
+    s
+}
+
+fn write_link_attr(w: &mut dyn Write, shared_libs: &Vec<String>) -> Result<()> {
+    for it in shared_libs.iter() {
+        writeln!(w, "#[link(name = \"{}\")]", get_library_name(it))?;
```

Comment:
I think this should also be used here: https://github.com/gtk-rs/gir/blob/a6737c72ae70ee7a17ae4ac7729905f9bb87e4e3/src/codegen/sys/cargo_toml.rs#L42

It's wrong what it does there currently.

Reply:
done